tests/5.0/declare_target/test_declare_target_device_type_nohost.F90

```diff
@@ -0,0 +1,97 @@
+!===--- test_declare_target_device_type_nohost.F90 -----------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! The declare target directive specifies that variables, functions(C,C++ and Fortran),
+! and subroutines (Fortran) are mapped to a device. If a device_type
+! clause is present on the contained declare target directive, then its argument 
+! determines which versions are made available. If device_type(nohost) is present 
+! only a device version of the procedure is made available. 
+! The device version of the function is specified via declare variant.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 10
+
+MODULE m_dat
+    !$omp declare target to(a, b, c, i)
+    INTEGER, DIMENSION(N) :: a, b, c
+    INTEGER :: i = 0
+END MODULE m_dat
+
+PROGRAM test_declare_target_device_type_nohost
+
+    USE iso_fortran_env
+    USE ompvv_lib
+    USE omp_lib
+    USE m_dat
+    implicit none
+    INTEGER :: errors = 0
+    
+    !$omp declare target to(target_function) device_type(nohost)
+    
+    OMPVV_TEST_OFFLOADING
+
+    ! initialize arrays on host
+    DO i = 1, N
+      a(i) = 0
+      b(i) = 0
+      c(i) = 0
+    END DO
+     
+    OMPVV_TEST_VERBOSE(test_device_type_nohost() .ne. 0)
+
+    OMPVV_REPORT_AND_RETURN()
+
+CONTAINS  
+    SUBROUTINE target_function()
+      DO i = 1, N
```

Comment:
Add "!$omp declare target to(target_function) device_type(nohost)" before the DO loop.

Reply:
It seems that we should not add the declare target directive here (see my comments above).

Reply:
Moved `!$omp declare target to(target_function) device_type(nohost)` into the internal procedure, `target_function`, as suggested.